### PR TITLE
Add bitbucket support to finding view SCM urls

### PR DESCRIPTION
**Pull request by drJabber**
**Description**
Adds support of bitbucket source code navigation URLs to SCM URLs in Finding views

**Test results**
Added tests in test_finding_model.py

**Documentation**
Documentation updated in  docs/content/en/integrations/source-code-repositories.md

### CODE REVIEW

**Comment by drJabber** [comment]
Hello, Team, could somebody approve PR? Although it modifies models.py it does not modify DB model itself - it just modifies Finding method, that composes SCM url to navigate to source code in browser, so I think it could be included into code base before v3 release.

File: dojo/settings/settings.dist.py
Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1471]: There is something wrong with your PR. Please fix your commit history.

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py
Comment (Maffooch) [reply dojo/settings/settings.dist.py:1471]: +1 ^

Comment (drJabber) [reply dojo/settings/settings.dist.py:1476]: fixed, thank you

Comment (manuel-sommer) [reply dojo/settings/settings.dist.py:1476]: No, it is not fixed.

Comment (drJabber) [reply dojo/settings/settings.dist.py:1476]: sorry, @manuel-sommer , what should i do to fix commit history? I thought I just should rebase to DefectDojo:dev ...
or may be should i squash my several commits into single?

Comment (manuel-sommer) [reply dojo/settings/settings.dist.py:1476]: This link will probably help: 
https://github.com/openedx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

Comment (drJabber) [reply dojo/settings/settings.dist.py]: tried to rebase as described in this manual, I hope now it's ok

**Review by Maffooch — APPROVED** [review]
Great job "feature flagging" this new behavior with the product meta

**Review by mtesauro — APPROVED** [review]
Approved

File: docs/content/en/integrations/source-code-repositories.md
Comment (cneill) [inline docs/content/en/integrations/source-code-repositories.md:30]:
```suggestion
If user does not set commit hash or branch/tag in appropriate fields of CI/CD Engagement edit form, the URL should look like in Interactive Engagement edit form.
```

File: docs/content/en/integrations/source-code-repositories.md
Comment (cneill) [inline docs/content/en/integrations/source-code-repositories.md:32]:
```suggestion
SCM navigation URL is composed from Repo URL using SCM Type. Github/Gitlab SCM type is default, but user could set certain SCM type in Product custom field "scm-type".
```

File: dojo/models.py
Comment (cneill) [inline dojo/models.py:3046]:
```suggestion
        # so to create browser url - git url should be recomposed like below:
```

File: dojo/models.py
Comment (cneill) [inline dojo/models.py:3053]:
```suggestion
        # hash or branch/tag should be appended to base browser link
```

**Review by cneill — CHANGES_REQUESTED** [review]
Just a few small things, otherwise looks good!

**Comment by drJabber** [comment]
> Just a few small things, otherwise looks good!

thanks, commited

**Review by manuel-sommer — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Comment by drJabber** [comment]
still need one more reviewer approval. Could anybody with write access review this, please?

**Comment by drJabber** [comment]
Hi @grendel513 / @devGregA / @blakeaowens, I'd like to ask for a review

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**
